### PR TITLE
Updates Compose Multiplatform to 1.7.0.

### PR DESCRIPTION
**Pull request by JayShortway**
As discussed in a weekly a few _weeks_ ago, [1.7.0 is finally stable](https://github.com/JetBrains/compose-multiplatform/releases/tag/v1.7.0). 

- Closes https://github.com/RevenueCat/purchases-kmp/issues/199
- Closes https://github.com/RevenueCat/purchases-kmp/issues/201

Also updates the `Podfile.lock`, because [that pipeline](https://app.circleci.com/pipelines/github/RevenueCat/purchases-kmp/770/workflows/ff5b9f7d-c644-4e03-b663-c259865e20bb/jobs/2386) seems to run too soon, before the new pod is actually published.

### CODE REVIEW

File: iosApp/Podfile.lock [hunk 1]
```diff
@@ -1,16 +1,16 @@
 PODS:
-  - PurchasesHybridCommon (13.3.0):
```
Comment (vegaro) [inline iosApp/Podfile.lock:2]: These changes seem unrelated. Maybe they come from local testing?

**Review by vegaro — APPROVED** [review]

**Review by tonidero — APPROVED** [review]

File: iosApp/Podfile.lock (hunk 1 above)
Comment (JayShortway) [reply iosApp/Podfile.lock:2]: See the PR description 😛 
>Also updates the `Podfile.lock`, because [that pipeline](https://app.circleci.com/pipelines/github/RevenueCat/purchases-kmp/770/workflows/ff5b9f7d-c644-4e03-b663-c259865e20bb/jobs/2386) seems to run too soon, before the new pod is actually published.

Comment (JayShortway) [reply iosApp/Podfile.lock:2]: Happy to extract that to a separate PR ofc!

**Review by vegaro — APPROVED** [review]
